### PR TITLE
Unstructured Vertical Solver

### LINKED ISSUE #413: Unstructured vertical solver
We should also have tests for vertical dependencies --> Write a simple vertical solver for unstructured grids on IIR *and* SIR level and check if the generated code is correct.

### PR DESCRIPTION
**Pull request by mroethlin**
## Technical Description

This PR adds a vertical solver using the naive ICON backend. This unstructured solver is generated twice: once from IIR and once from SIR using the Python interface. 

### Resolves / Enhances

resolves #413 

### Testing

* The stencil generated form IIR is tested against a manufactured solution (added to the atlas-integration tests in `dawn`)
* The stencil generated from SIR is added to the post installation step of dawn. The generated stencil was tested manually against the same manufactured solution. No automatic testing of the stencil built from SIR is done, except for compiling it during the post install step

### CODE REVIEW

**Comment by mroethlin** [comment]
launch jenkins

**Comment by mroethlin** [comment]
launch jenkins

File: gtclang/test/atlas-integration-test/GenerateUnstructuredStencils.cpp [hunk 1]
```diff
@@ -48,10 +49,14 @@ int main() {
...
-                        b.stage(b.vregion(dawn::sir::Interval::Start, dawn::sir::Interval::End,
-                                          b.stmt(b.assignExpr(b.at(out_f), b.at(in_f))))))));
+        "copyCell",
+        b.stencil(
+            b.multistage(LoopOrderKind::Parallel,
+                         b.stage(b.vregion(dawn::sir::Interval::Start, dawn::sir::Interval::End,
+                                           b.stmt(b.assignExpr(b.at(out_f), b.at(in_f)))))),
+            b.multistage(LoopOrderKind::Parallel,
+                         b.stage(b.vregion(dawn::sir::Interval::Start, dawn::sir::Interval::End,
+                                           b.stmt(b.assignExpr(b.at(out_f), b.at(in_f))))))));
```
Comment (Stagno) [inline gtclang/test/atlas-integration-test/GenerateUnstructuredStencils.cpp]: Is this duplicated by mistake?

File: gtclang/test/atlas-integration-test/GenerateUnstructuredStencils.cpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 //===------------------------------------------------------------------------------------------===//
 
 #include "dawn/CodeGen/CXXNaive-ico/CXXNaiveCodeGen.h"
+#include "dawn/CodeGen/CXXNaive/CXXNaiveCodeGen.h"
```
Comment (Stagno) [inline gtclang/test/atlas-integration-test/GenerateUnstructuredStencils.cpp]: Looks like it's not actually needed, or am I missing something?

File: dawn/examples/python/tridiagonal_solve_unstructured.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+dawn.dawnTranslationUnitGetStencil.restype = c_void_p
+dawn.dawnTranslationUnitGetStencil.argtypes = [
+    ctypes.c_void_p,
+    ctypes.c_char_p
+]
+code = dawn.dawnTranslationUnitGetStencil(tu, b_stencilName)
+
+# write to file
+f = open(os.path.dirname(os.path.realpath(__file__))
+         + "/data/unstructured_tridiagonal_solve.cpp", "w")
```
Comment (Stagno) [inline dawn/examples/python/tridiagonal_solve_unstructured.py]: Is this later compared to a reference code?

File: dawn/examples/python/tridiagonal_solve_unstructured.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    # create the out = in[i+1] statement
+    body_ast = make_ast(
+        [make_assignment_stmt(
+            make_unstructured_field_access_expr("d"),
+            make_binary_operator(
+                make_unstructured_field_access_expr("c"),
+                "*",
+                make_unstructured_field_access_expr("d", make_unstructured_offset(False), 1)
+            ),
+            "-="
```
[the remark this answers is not in the document]
Comment (mroethlin) [reply dawn/examples/python/tridiagonal_solve_unstructured.py]: re-tested, works

**Comment by mroethlin** [comment]
launch jenkins

**Pull request merged**